### PR TITLE
Use json for writing floats as strings

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

This is the change that would be required by https://github.com/open-telemetry/opentelemetry-specification/pull/2343.  It changes how small floats are represented by using the json marshaller.  The test would fail with the current implementation, as it returns ".000000009" instead of "9e-9".

### CODE REVIEW

File: model/pdata/common.go
Comment (bogdandrutu) [inline model/pdata/common.go:383]: Why? Should we use what json.Marshal uses instead? The reason is I believe json.Marshal is extremely slow because of reflection etc.

Comment (dashpole) [reply model/pdata/common.go:383]: We can do that.  It would involve copying https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/encoding/json/encode.go;l=585.

Comment (dashpole) [reply model/pdata/common.go]: Done. Benchmark improvement:

With `json.Marshal`:

`BenchmarkAttributeValueFloat_AsString-16    	 3270664	       364.8 ns/op`

With the new implementation:

`BenchmarkAttributeValueFloat_AsString-16    	 6703143	       174.0 ns/op`

I'll let you decide if the change is worth it :).

**Comment by dashpole** [comment]
This is ready for review

File: model/pdata/common.go
Comment (bogdandrutu) [reply model/pdata/common.go]: As most likely a future maintainer of the codebase what would be your preference 😁?

Comment (dashpole) [reply model/pdata/common.go]: I think its probably worth it.  We risk drifting from the upstream implementation, but it hasn't changed in years, so we are probably safe.  The benchmark differences are fairly large.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@dashpole maybe rebase to re-trigger checks?

**Comment by dashpole** [comment]
pushed again

**Comment by bogdandrutu** [comment]
@dashpole not sure what happens,.... maybe open a new PR from a cloned branch?

**Comment by dashpole** [comment]
it woke up!

**Comment by dashpole** [comment]
rebased

**Pull request merged**
